### PR TITLE
Added 30 plant diseases to a new table using a web API to provide the necessary data.

### LINKED ISSUE #354: Create a new table to record plant diseases and import data from an external source

### PR DESCRIPTION
**Pull request by captainbraddles**
Added 30 plant diseases to a new table using a web API to provide the necessary data. This closes #354

### CODE REVIEW

**Comment by SapphicFire** [comment]
Please note that the issue closing keywords work when entered in the PR description, so you may want to put the text currently in the title in the description and update the title to be more relevant

File: 9c697d609329ca10c7067a418bba1022/update/sys_rest_message_a671b6e1975512109790fefce053af17.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        <access>package_private</access>
+        <authentication_type>no_authentication</authentication_type>
+        <basic_auth_password/>
+        <basic_auth_profile/>
+        <basic_auth_user/>
+        <description/>
+        <name>Plant disease import</name>
+        <oauth2_profile/>
+        <protocol_name/>
+        <rest_endpoint>https://perenual.com/api/pest-disease-list?key=sk-oq9f6710caae7048c7302</rest_endpoint>
```
Comment (SapphicFire) [inline 9c697d609329ca10c7067a418bba1022/update/sys_rest_message_a671b6e1975512109790fefce053af17.xml:12]: This contains your API Key - please remove it and make it a query parameter instead

File: 9c697d609329ca10c7067a418bba1022/update/sys_rest_message_fn_4c913aa1975512109790fefce053af8e.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        <basic_auth_profile/>
+        <basic_auth_user/>
+        <content/>
+        <function_name>Get Plant Diseases</function_name>
+        <http_method>get</http_method>
+        <lock>false</lock>
+        <oauth2_profile/>
+        <protocol_name/>
+        <qualified_name>Plant disease import [ Get Plant Diseases ]</qualified_name>
+        <rest_endpoint>https://perenual.com/api/pest-disease-list?key=sk-oq9f6710caae7048c7302</rest_endpoint>
```
Comment (SapphicFire) [inline 9c697d609329ca10c7067a418bba1022/update/sys_rest_message_fn_4c913aa1975512109790fefce053af8e.xml:14]: Also holds your API key

File: 9c697d609329ca10c7067a418bba1022/update/sys_script_include_0bb58e71971112109790fefce053af28.xml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        <mobile_callable>false</mobile_callable>
+        <name>get_plant_diseases</name>
+        <sandbox_callable>false</sandbox_callable>
+        <script><![CDATA[var get_plant_diseases = Class.create();
+get_plant_diseases.prototype = {
+    initialize: function() {
+    },
+
+	getPlantDiseases: function(lat, lon) {
+        var diseases = new sn_ws.RESTMessageV2('Plant disease import', 'Get Plant Diseases');
```
Comment (SapphicFire) [inline 9c697d609329ca10c7067a418bba1022/update/sys_script_include_0bb58e71971112109790fefce053af28.xml:18]: Add a property in the global scope that holds the API key for this platform and call it from this script include and provide it as a query parameter

File: 9c697d609329ca10c7067a418bba1022/update/sys_script_include_0bb58e71971112109790fefce053af28.xml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+			var diseaseRecord = new GlideRecord('x_snc_plant_mgmt_plant_diseases');
+			diseaseRecord.initialize();
+			diseaseRecord.common_name = commonName;
+			diseaseRecord.scientific_name = scientificName;
+			diseaseRecord.family = family;
+			diseaseRecord.description = description;
+			diseaseRecord.solution = solution;
+			diseaseRecord.host = host;
+			diseaseRecord.insert();
```
Comment (SapphicFire) [inline 9c697d609329ca10c7067a418bba1022/update/sys_script_include_0bb58e71971112109790fefce053af28.xml:61]: Consider updating existing if values match such as common or scientific name so you avoid duplicates

**Review by SapphicFire — CHANGES_REQUESTED** [review]
Awesome PR! There are a few things to tidy up, such as removing your API key and handling duplicates as there are some duplicate diseases. We will leave this open so you can make those updates.

File: 9c697d609329ca10c7067a418bba1022/update/sys_rest_message_a671b6e1975512109790fefce053af17.xml (hunk 1 above)
Comment (captainbraddles) [reply 9c697d609329ca10c7067a418bba1022/update/sys_rest_message_a671b6e1975512109790fefce053af17.xml]: Hi @SapphicFire. I have removed the API key and used a query parameter instead.

File: 9c697d609329ca10c7067a418bba1022/update/sys_rest_message_fn_4c913aa1975512109790fefce053af8e.xml (hunk 2 above)
Comment (captainbraddles) [reply 9c697d609329ca10c7067a418bba1022/update/sys_rest_message_fn_4c913aa1975512109790fefce053af8e.xml]: Hi @SapphicFire. I have removed the API key and used a query parameter instead.

File: 9c697d609329ca10c7067a418bba1022/update/sys_script_include_0bb58e71971112109790fefce053af28.xml (hunk 3 above)
Comment (captainbraddles) [reply 9c697d609329ca10c7067a418bba1022/update/sys_script_include_0bb58e71971112109790fefce053af28.xml]: Hi @SapphicFire. I have removed the API key and used a query parameter instead.

File: 9c697d609329ca10c7067a418bba1022/update/sys_script_include_0bb58e71971112109790fefce053af28.xml (hunk 4 above)
Comment (captainbraddles) [reply 9c697d609329ca10c7067a418bba1022/update/sys_script_include_0bb58e71971112109790fefce053af28.xml]: Hi @SapphicFire. I have updated the duplicate names to include the scientific name where the common name is the same, but the scientific is different. I also removed a genuine duplicate.

**Comment by captainbraddles** [comment]
Hi @SapphicFire. Apologies, I'm not a regular git/github user. I think I've been able to add everything in to the latest commit now.

**Review by SapphicFire — APPROVED** [review]
Looks good to me, thank you for working with me to get it all together!

**Pull request merged**